packages/dd-trace/src/plugins/util/web.js

```diff
@@ -257,21 +258,28 @@ const web = {
   },
 
   // Extract the parent span from the headers and start a new span as its child
-  startChildSpan (tracer, name, req) {
+  startChildSpan (tracer, name, req, traceCtx) {
     const headers = req.headers
-    const context = contexts.get(req)
+    const reqCtx = contexts.get(req)
     let childOf = tracer.extract(FORMAT_HTTP_HEADERS, headers)
 
+    const store = storage('legacy').getStore()
+
     // we may have headers signaling a router proxy span should be created (such as for AWS API Gateway)
     if (tracer._config?.inferredProxyServicesEnabled) {
-      const proxySpan = createInferredProxySpan(headers, childOf, tracer, context)
+      const proxySpan = createInferredProxySpan(headers, childOf, tracer, reqCtx, store, traceCtx)
       if (proxySpan) {
         childOf = proxySpan
       }
     }
 
     const span = tracer.startSpan(name, { childOf, links: childOf?._links })
 
+    if (traceCtx) {
```

Comment:
Why can't we rely on `startSpan` for this?

Reply:
Ah yes I guess we'd need to have some WebPlugin class instead of this helper, or switch everything to composition. For now it's fine then, we can figure this out later.